### PR TITLE
Add Artifactory support

### PR DESCRIPTION
**Pull request by thommay**
Signed-off-by: Thom May <thom@chef.io>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ branches:
 
 language: ruby
 rvm:
-  - 2.1
```
Comment (tas50) [inline .travis.yml]: Pour one out

**Comment by coderanger** [comment]
Maybe I'm misunderstanding how omnifetch works, but I don't see any authentication in here?

**Comment by coderanger** [comment]
Ahh, I think I see, ChefDK is passing in a concrete HTTP client object with the auth header already configured.

**Comment by thommay** [comment]
Yeah, the http object gets the auth and is DI'd in here.

And yes, there is definitely enough reuse that we can think about pushing some bits down in to the base class.

**Pull request merged**